src/builder/builder.js

```diff
@@ -62,10 +62,7 @@ module.exports = function (createChunker, ipldResolver, createReducer, _options)
     waterfall([
       (cb) => DAGNode.create(d.marshal(), cb),
       (node, cb) => {
-        ipldResolver.put({
-          node: node,
-          cid: new CID(node.multihash)
-        }, (err) => cb(err, node))
+        ipldResolver.put(node, new CID(node.multihash), (err) => cb(err, node))
```

Comment:
How about

```js
resolver.put(node, {cid: cid}, callback)
resolver.put(node, {format: 'dag-pb', hash: 'sha2-256'}, callback)

// or this could work as well, by doing an `Cid.isCid(arguments[0])` check

resolver.put(cid, node, callback)
resolver.put(node, {format: 'dag-pb', hash: 'sha2-256'}, callback)
```

I am not a 100% sure about these, haven't had time to think through them, but they seem to be better than the current version to me.